src/ServiceControl/Notifications/Email/SendEmailNotificationHandler.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace ServiceControl.Notifications.Email
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Logging;
+    using NServiceBus.Transport;
+    using Raven.Client;
+    using ServiceBus.Management.Infrastructure.Settings;
+
+    public class SendEmailNotificationHandler : IHandleMessages<SendEmailNotification>
+    {
+        readonly IDocumentStore store;
+        readonly EmailThrottlingState throttlingState;
+
+        public SendEmailNotificationHandler(IDocumentStore store, Settings settings, EmailThrottlingState throttlingState)
+        {
+            this.store = store;
+            this.throttlingState = throttlingState;
+
+            emailDropFolder = settings.EmailDropFolder;
+        }
+
+        public async Task Handle(SendEmailNotification message, IMessageHandlerContext context)
+        {
+            NotificationsSettings notifications;
+
+            using (var session = store.OpenAsyncSession())
```

Comment:
would this be a good case for https://ravendb.net/docs/article-page/3.5/csharp/client-api/how-to/setup-aggressive-caching ? Especially because `SaveChangesAsync` will invalidate the cache https://ravendb.net/docs/article-page/3.5/csharp/client-api/how-to/setup-aggressive-caching#invalidate-on-savechanges

Reply:
done